test/scripts/integration-tests.sh

```
@@ -0,0 +1,304 @@
+
+    # Validate PID is a number
+    if [[ ! "$server_pid" =~ ^[0-9]+$ ]]; then
+        echo "❌ Failed to get valid server PID: '$server_pid'"
+        cat "$LOG_FILE"
+        exit 1
+    fi
+
+    echo "📝 Captured server PID: $server_pid"
+
+    # Ensure server is stopped on exit
+    trap "cleanup $server_pid" EXIT INT TERM
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)